chrome/browser/android/adblock/regex_resource_type_detector.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * This file is part of Adblock Plus <https://adblockplus.org/>,
+ * Copyright (C) 2006-present eyeo GmbH
+ *
+ * Adblock Plus is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * Adblock Plus is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Adblock Plus.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef CHROME_BROWSER_ANDROID_REGEX_RESOURCE_TYPE_DETECTOR_H_
+#define CHROME_BROWSER_ANDROID_REGEX_RESOURCE_TYPE_DETECTOR_H_
+
+#include <string>
+#include "content/public/common/resource_type.h"
+
+class RegexResourceTypeDetector {
```

Comment:
IMO we don't need a class here, mere a method in e.g. `AdblockBridge` namespace would be enough.

Reply:
`AdblockBridge` class originally was created just to provide communication over JNI between java and c++ classes (that's why it's `..Bridge`). BTW it has no namespace but probably should (this was meant to be done while clean-up. We have this task in todo-list).

I'd prefer to keep them separated as it will be too much of "job" for `AdblockBridge` and later i'd like to have unit-test for `RegexResouceTypeDetector` class.

Reply:
Still, there is no need for a class, mere a function is enough.

Reply:
as stated below, I agree with Sergei.